src/main/java/Duke.java

```diff
@@ -1,10 +1,140 @@
+import exceptions.DukeException;
+import exceptions.DukeTaskNoDescException;
+import exceptions.DukeUnknownTaskException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+
+    public static void createList(ArrayList<Task> listArr) {
+        System.out.println("Here are the tasks in your list:");
+        for (int j = 0; j < listArr.size(); j++) { // printing out all items in the list
+            System.out.println(j + 1 + ". " + listArr.get(j));
+        }
+    }
+
+    public static void createTodo(String userTxt, ArrayList<Task> listArr) {
+        String test = userTxt.substring(4);
+        try {
+            if (test.strip().equals("")) {
+                throw new DukeTaskNoDescException();
+            } else {
+                Task t = new ToDo(userTxt.substring(5));
+                listArr.add(t); // add task to list
+                System.out.println("Got it. I've added this task:");
+                System.out.println(t);
+                System.out.println("Now you have " + listArr.size() + " tasks in the list.");
+            }
+        } catch (DukeException ex) {
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    public static void createDeadline(String userTxt, ArrayList<Task> listArr) {
+        String test = userTxt.substring(8);
+        try {
+            if (test.strip().equals("")) {
+                throw new DukeTaskNoDescException();
+            } else {
+                int splitIdx = userTxt.lastIndexOf("/");
```

Comment:
Should this be `"/by"` instead of `"/"`? (similar to how `createEvent` does it) 
You could also consider using the `stripTrailing()` method on strings to remove unecessary whitespace!